libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -242,6 +242,16 @@ func (k *kubernetesAnnotator) Run(event *beat.Event) (*beat.Event, error) {
 	return event, nil
 }
 
+func (k *kubernetesAnnotator) Close() error {
+	if k.watcher != nil {
+		k.watcher.Stop()
```

Comment:
Good catch, the watcher actually uses the cache. But I don't think this is a problem *by now*, the cache can be used even after being stopped, it only doesn't clean expired entries.

Changing the watcher to use context cancellation instead of Start/Stop will require some refactors.

I will give a try to improve this, let me know if you don't consider it so blocking :slightly_smiling_face:

Reply:
> Good catch, the watcher actually uses the cache.

Sorry, I was wrong, the informer used by `add_kuberentes_metadata` has a cache, but is a different one, the cache stopped here is not used by the watcher.